examples/chip-tool/commands/pairing/PairingCommand.cpp

```diff
@@ -222,6 +221,45 @@ CHIP_ERROR PairingCommand::AddWiFiNetwork()
     return mCluster.AddWiFiNetwork(successCallback, failureCallback, mSSID, mPassword, kBreadcrumb, kTimeoutMs);
 }
 
+chip::ByteSpan PairingCommand::GetThreadNetworkId() const
```

Comment:
We have a class, `Thread::ThreadOperationalDataset`, for this in `lib/support/ThreadOperationalDataset.h`.

Can that be used to provide this parser?

Reply:
that class was exactly what the first 2 versions of this PR used but it had limitations and required extra storage: local storage had the problem of going out of scope when used, global storage was also questioned and indeed unnecessary.

Reply:
So I can see that `Thread:: ThreadOperationalDataset` doesn't give a zero-copy way to instantiate an instance from a `bytespan`.  But why not just make `mOperationalDataset` of this type in the first place?

Short of that, there still seem to be other reuse opportunities.  For instance, if `mOperationalDataset` can't be of type `Thread:: ThreadOperationalDataset`, why not construct one of these locally?  And if that isn't possible, why can't this parser at least use the `Thread:: ThreadTLV` type?

Reply:
Yes we could make mOperationalDataset of type ThreadOperationalDataset but it would only be useful if the API allowed ByteSpan accessors to its internal storage, otherwise we would be back to where we started, needing global or member storage for something that does not logically need it. Note that previous 2 versions of this PR used a local instance of ThreadOperationalDataset and was suffering from same usability issues. Regarding why not at least ThreadTLV type it's very simple: because it is private and cannot be used from the outside.

Reply:
I share the concern. ThreadOperationalDataset is our own utility class; we can change it.

Reply:
I switched back to using Thread OperationalDataset class as you requested and provided storage for the extendedPanId as a member variable, this way everyone's request can be accommodated.